doc/src/manual/missing.md

```diff
@@ -254,6 +254,14 @@ julia> Array{Union{Missing, String}}(missing, 2, 3)
  missing  missing  missing
 ```
 
+!!! note
+    Currently for `T` that is bits type (`isbitstype(T)` returns `true`),
+    `Array{Union{Missing, T}}(undef, dims)` creates an array filled with
+    missing values. Also calling `similar` to create an unitialized array that
+    has element type of the form `Union{Missing, T}` creates an array filled with
+    `missing` if `T` is bits type. This behavior is an implementation detail
```

Comment:
"Implementation detail that may change in the future" might be a bit strong, as it makes it sound like the public API cannot be relied on. Maybe just recommend using `Array{Union{Missing, T}}(missing, dims)`, noting that it works for all types? Maybe also remove the leading "currently".

Reply:
I know it is strong, but from the discussion I understood that we want to be discouraging. The question is if in e.g. Julia 1.7 we will guarantee this behavior (I know we rely on it internally, and the probability that this will change is minimal, but theoretically it is possible). In general - I understand that this is only the current behavior but it is not a part of a **public** API.

Actually, I have written this PR is exactly to clarify: do we want to make it a public API or just describe the current behavior and warn users that this is not a part of public API?

Reply:
AFAIK, that's part of the public API (anyway, any behavior that is stable like this will de facto end up as being part of the API since lots of code will rely on it). We can mark this for triage to make sure that's the case.

Reply:
OK - can you please mark it?

> any behavior that is stable like this will de facto end up as being part of the API since lots of code will rely on it

Yes- but I would prefer an explicit decision 😄.

Reply:
If you have `Union{SingletonType, NonSingletonType}` you are guaranteed to get the singleton type (e.g. `Missing`), and that is guaranteed. For other combinations it's less predictable.